File: Two Pointers/0633.Sum of Square Numbers/Solution.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        i, j = 0, int(math.sqrt(c))
        while i <= j:
            cur = i * i + j * j
            if cur == c:
                return True
            if cur > c:
                j -= 1
            else:
                i += 1
        return False

File: Two Pointers/0633.Sum of Square Numbers/Solution.java
class Solution {
    public boolean judgeSquareSum(int c) {
        int i = 0, j = (int) Math.sqrt(c);
        while (i <= j) {
            long cur = (long) i * i + (long) j * j;
            if (cur == c) {
                return true;
            }
            if (cur > c) {
                j--;
            } else {
                i++;
            }
        }
        return false;
    }
}